File: Structural-Patterns/Bridge/C#/Bridge/Abstraction/OneTimePayment.cs
public class OneTimePayment(IPaymentGateway gateway, double amount) : Payment(gateway) {

    private double amount = amount;
    private String transactionId = "";

    public override void MakePayment() {
        Console.WriteLine("Initiating a one-time purchase of Rs." + amount);
        paymentGateway.ProcessPayment(amount);
        this.transactionId = "txn" + DateTime.Now.Ticks;
        Console.WriteLine("One-time purchase complete. Transaction ID: " + this.transactionId);
    }

    public override void IssueRefund() {
        paymentGateway.RefundPayment(transactionId, amount);
        Console.WriteLine("Initiating refund for one time payment of amount : " + amount);
    }
}

File: Structural-Patterns/Bridge/Java/Abstraction/OneTimePayment.java
package Abstraction;

import Implementation.IPaymentGateway;

public class OneTimePayment extends Payment {

    private double amount;
    private String transactionId;

    public OneTimePayment(IPaymentGateway gateway, double amount){
        super(gateway);
        this.amount = amount;
    }

    @Override
    public void makePayment() {
        System.out.println("Initiating a one-time purchase of Rs." + amount);
        paymentGateway.processPayment(amount);
        this.transactionId = "txn" + System.currentTimeMillis();
        System.out.println("One-time purchase complete. Transaction ID: " + this.transactionId);
    }

    @Override
    public void issueRefund() {
        paymentGateway.refundPayment(transactionId, amount);
        System.out.println("Initiating refund for one time payment of amount : " + amount);
    }
}
